sqlalchemy-stubs/sql/dml.pyi

```diff
@@ -57,27 +59,26 @@ class UpdateBase(
     __visit_name__: str = ...
     named_with_column: bool = ...
     is_dml: bool = ...
-    def params(self, *arg: Any, **kw: Any) -> None: ...
-    def with_dialect_options(self, **opt: Any) -> None: ...
-    def bind(self): ...
     bind: Any = ...
-    def returning(self, *cols: Any) -> None: ...
+    def params(self: _UB, *arg: Any, **kw: Any) -> _UB: ...
```

Comment:
this actually raises not implemented. so maybe it would return no-return?

Reply:
It does, but subclasses implement this as generative, so it's better to have the right return type that subclasses should implement so we don't have to use a `# type: ignore[override]`. As an example:

```python
from typing import NoReturn

class One:
    def go_one(self) -> int:
        raise NotImplementedError()

    def go_two(self) -> NoReturn:
        raise NotImplementedError()

class Two(One):
    def go_one(self) -> int:
        return 1

    def go_two(self) -> int:    # incompatible return type
        return 2
```

Reply:
understood, thanks.

I could not find any subclass that implements it though. 
Looking at the error message, maybe the issue is the opposite? ie does this raise?
```py
class One:
    def go_one(self) -> int:
        raise NotImplementedError()

    def go_two(self) -> int:
        return 42

class Two(One):
    def go_one(self) -> int:
        return 1

    def go_two(self) -> NoReturn:    # incompatible return type?
        raise NotImplementedError()
```

Reply:
You're right, I got my scenarios mixed up. `ClauseElement.params()` returns a copy of itself, so `UpdateBase.params()` needs to indicate that it returns a copy of itself (even though it raises).